macro/src/dialect/operation/builder.rs

```diff
@@ -1,82 +1,99 @@
-struct TypeStateList(Vec<TypeStateItem>);
+struct TypeStateList {
+    items: Vec<TypeStateItem>,
+    unset: GenericArgument,
+    set: GenericArgument,
+}
 
 impl TypeStateList {
-    pub fn iter(&self) -> impl Iterator<Item = &TypeStateItem> {
-        self.0.iter()
+    pub fn new(items: Vec<TypeStateItem>) -> Self {
+        Self {
+            items,
+            unset: syn::parse2(quote!(::melior::dialect::ods_support::Unset)).unwrap(),
+            set: syn::parse2(quote!(::melior::dialect::ods_support::Set)).unwrap(),
+        }
     }
 
-    pub fn iter_any(&self) -> impl Iterator<Item = &Ident> {
-        self.0.iter().map(|item| &item.any)
+    pub fn items(&self) -> impl Iterator<Item = &TypeStateItem> {
+        self.items.iter()
     }
 
-    pub fn iter_any_without<'a>(
+    pub fn parameters(&self) -> impl Iterator<Item = &GenericArgument> {
+        self.items().map(|item| &item.generic_param)
+    }
+
+    pub fn parameters_without<'a>(
         &'a self,
         field_name: &'a str,
-    ) -> impl Iterator<Item = &Ident> + '_ {
-        self.0.iter().filter_map(move |item| {
-            if item.field_name != field_name {
-                Some(&item.any)
-            } else {
-                None
-            }
-        })
+    ) -> impl Iterator<Item = &GenericArgument> + '_ {
+        self.items()
+            .filter(move |item| item.field_name != field_name)
+            .map(|item| &item.generic_param)
     }
 
-    pub fn iter_set_yes<'a>(&'a self, field_name: &'a str) -> impl Iterator<Item = &Ident> + '_ {
-        self.0.iter().map(move |item| {
+    pub fn arguments_replace<'a>(
+        &'a self,
+        field_name: &'a str,
+        argument: &'a GenericArgument,
+    ) -> impl Iterator<Item = &GenericArgument> + '_ {
+        self.items().map(move |item| {
             if item.field_name == field_name {
-                &item.yes
+                argument
             } else {
-                &item.any
+                &item.generic_param
             }
         })
     }
 
-    pub fn iter_set_no<'a>(&'a self, field_name: &'a str) -> impl Iterator<Item = &Ident> + '_ {
-        self.0.iter().map(move |item| {
-            if item.field_name == field_name {
-                &item.no
-            } else {
-                &item.any
-            }
-        })
+    pub fn arguments_set<'a>(
+        &'a self,
+        field_name: &'a str,
+    ) -> impl Iterator<Item = &GenericArgument> + '_ {
+        self.arguments_replace(field_name, &self.set)
     }
 
-    pub fn iter_yes(&self) -> impl Iterator<Item = &Ident> {
-        self.0.iter().map(|item| &item.yes)
+    pub fn arguments_unset<'a>(
+        &'a self,
+        field_name: &'a str,
+    ) -> impl Iterator<Item = &GenericArgument> + '_ {
+        self.arguments_replace(field_name, &self.unset)
+    }
+
+    pub fn arguments_all_set(&self) -> impl Iterator<Item = &GenericArgument> {
+        repeat(&self.set).take(self.items.len())
     }
```

Comment:
Can we generalize these `arguments_all_set` and `arguments_all_unset` functions a little more?

```suggestion
    pub fn arguments_all(&self, argument: &GenericArgument) -> impl Iterator<Item = &GenericArgument> {
        repeat(argument).take(self.items.len())
    }
```